src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs

```diff
@@ -1513,5 +1540,18 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                 throw new Exception("Background Exception");
             }
         }
+
+        private class AsyncWaitingService : BackgroundService
+        {
+            public AsyncWaitingService() { }
+
+            protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+            {
+                while (!stoppingToken.IsCancellationRequested)
+                {
+                    await Task.Delay(1000, stoppingToken);
+                }
```

Comment:
```C#
await Task.Delay(-1, stoppingToken);
```
?  Just wondering what purpose the loop serves.

Reply:
It simulates the "worker" template. See https://github.com/dotnet/aspnetcore/pull/35127/files.

It just doesn't have the `_logger.LogInformation("Worker running at: {time}", DateTimeOffset.Now);` part.

Reply:
If it's important that it mirror that structure, I'd suggest adding it as a comment, so that someone like me doesn't come along later and clean it up ;-)

Reply:
Fixed. Thanks for the feedback.